dex/farm-with-locked-rewards/src/lib.rs

```diff
@@ -0,0 +1,331 @@
+        owner: ManagedAddress,
+        admins: MultiValueEncoded<ManagedAddress>,
+    ) {
+        self.base_farm_init(
+            reward_token_id,
+            farming_token_id,
+            division_safety_constant,
+            owner,
+            admins,
+        );
+
+        self.penalty_percent().set_if_empty(DEFAULT_PENALTY_PERCENT);
+        self.minimum_farming_epochs()
+            .set_if_empty(DEFAULT_MINUMUM_FARMING_EPOCHS);
+        self.burn_gas_limit().set_if_empty(DEFAULT_BURN_GAS_LIMIT);
+        self.pair_contract_address().set(&pair_contract_address);
+    }
+
+    #[payable("*")]
+    #[endpoint(enterFarm)]
+    fn enter_farm(&self) -> EnterFarmResultType<Self::Api> {
+        let caller = self.blockchain().get_caller();
+        self.is_address_whitelisted(&caller);
+        let payments = self.call_value().all_esdt_transfers();
+        let base_enter_farm_result = self.enter_farm_base(
+            payments,
+            Self::generate_aggregated_rewards_with_boosted_yields,
+            Self::default_create_enter_farm_virtual_position,
+            Self::get_default_merged_farm_token_attributes,
+            Self::create_farm_tokens_by_merging,
+        );
+
+        let output_farm_token_payment = base_enter_farm_result.new_farm_token.payment;
+        self.send_payment_non_zero(&caller, &output_farm_token_payment);
+
+        output_farm_token_payment
+    }
+
+    #[payable("*")]
+    #[endpoint(claimRewards)]
+    fn claim_rewards(&self) -> ClaimRewardsResultType<Self::Api> {
+        let caller = self.blockchain().get_caller();
+        self.is_address_whitelisted(&caller);
+        let calculate_reward_fn = |sc_ref: &Self,
+                farm_token_amount: &BigUint,
+                attributes: &FarmTokenAttributes<Self::Api>,
+                storage_cache: &StorageCache<Self>| {
+            Self::calculate_reward_with_boosted_yields(
+                sc_ref, 
+                &caller, 
+                farm_token_amount, 
+                attributes, 
+                storage_cache
+            )
+        };
+
+        let payments = self.call_value().all_esdt_transfers();
+        let base_claim_rewards_result = self.claim_rewards_base(
+            payments,
+            Self::generate_aggregated_rewards_with_boosted_yields,
+            calculate_reward_fn,
+            Self::default_create_claim_rewards_virtual_position,
+            Self::get_default_merged_farm_token_attributes,
+            Self::create_farm_tokens_by_merging,
+        );
+
+        let output_farm_token_payment = base_claim_rewards_result.new_farm_token.payment;
+        let rewards_payment = base_claim_rewards_result.rewards;
+        self.send_payment_non_zero(&caller, &output_farm_token_payment);
+        self.send_to_lock_contract_non_zero(caller, rewards_payment.token_identifier.clone(), rewards_payment.amount.clone());
+
+        (output_farm_token_payment, rewards_payment).into()
```

Comment:
This should return the locked tokens as output instead of the initial unlocked rewards (i.e. should return the output of `send_to_lock_contract_non_zero`, and not `rewards_payment` directly)